src/lib/resources/record.resource.ts

```diff
@@ -54,4 +56,33 @@ export class RecordResource extends BaseResource {
 
     return this.getVoFromResponse<RecordResponse>(response, 'RecordVO');
   }
+
+  public async addStorage(record: RecordVO) {
+    const accountResponse = await this.api.account.getSessionAccount();
+    const account = accountResponse.Results[0].data[0].AccountVO;
+    const billingResponse = await this.api.billing.addStorage(
+      account.accountId,
+      record.size
+    );
+    return this.getVoFromResponse<BillingResponse>(
+      billingResponse,
+      'LedgerNonfinancialVO'
+    );
+  }
+
+  public async getPresignedUrl(fileType: string, record: RecordVO) {
+    const response = await this.api.record.getPresignedUrl(fileType, record);
+    if (!response.isSuccessful) {
+      throw new PermSdkError(
+        'could not get presigned url',
+        response.Results[0].message
+      );
+    }
+    return this.getVoFromResponse<SimpleVOResponse>(response, 'SimpleVO');
+  }
+
+  public async registerRecord(record: RecordVO, s3url: string) {
+    const response = await this.api.record.registerRecord(record, s3url);
+    return this.getVoFromResponse<RecordResponse>(response, 'RecordVO');
+  }
```

Comment:
This is a faithful recreation of the back-end API, and a reasonable first pass, but I think we can do better! As is, we're putting a large burden on the client of our SDK. Client code needs to call `getPresignedUrl`, interpret the response to make a call to S3, and then call `registerRecord`. That's a lot of boilerplate code, and not only that, users of our SDK need to know how to do all of those things!

Instead, I would hope to see something more like the `uploadFromUrl` method in this class: given a file[1], and much of the record details (archive, filename, display name, description, parent folder), this method should handle all of the steps and either pass or fail.

[1] well, ideally, a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) (of which [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) is a subtype), so that we can pass in-memory data without having to go through the filesystem

Reply:
I totally agree that making this simpler is a good change, but I would say that's a followup PR.  If you look at our notes from April 8th, we agreed that we'd follow this process, which is why I did it this way.

Reply:
Ah, I'd forgotten that we discussed that - thanks!